### ISSUE TITLE
status_at_link

### ISSUE THREAD

**Issue by SiccarPoint** [landlab/landlab#340]
Status at link should become writable. This will make the BC bookkeeping harder, but is worth it.

At the moment, inactive links can only be produced by the judicious setting of CLOSED_NODEs. However, we should definitely allow the following:

```
6 core nodes, 6 active links, one inactive link:
* ----- * ----- *
|       X       |
|       X       |
* ----- * ----- *

```

i.e., as long as not all links at a node are not active, a node can still be core.

**Comment by gregtucker** [comment]
I suggest proceeding with caution here. The example above would actually change the definition of an inactive link from:
NOW: any link that touches one or more closed-boundary nodes OR connects two boundaries of any type
PROPOSED (if I understand right): either the above OR any link that a programmer has decided ought to be inactive.

**Comment by SiccarPoint** [comment]
Yes, this is our current de facto definition (though I’m not sure how many places it’s actually spelled out - I’ll have to check). But an actual process definition of an inactive link is one that cannot carry flux, in the same way that a closed node is one that can’t receive flux. At the moment, this isn’t quite true, for the reasons this issue is getting at. It would be nice to have out boundary conditions actually have robust physical meaning, as well as robust grid-based definitions.

On this basis, I think we would be good to modify our definitions. But you’re right, we need to take some care. More input on this would be very welcome.

(Reading this back, I’m not sure this 100% makes sense. Let me know if I need to be clearer).

**Comment by mcflugen** [comment]
@SiccarPoint: is what you are proposing just a matter of convenience for users or is it true that sometimes you can't get the same results but carefully setting node statuses. Physically, why would you want to connect two core nodes with an inactive link? Or, conversely, why do those nodes have to be core?

**Comment by SiccarPoint** [comment]
This mainly arose because I was trying to demonstrate something specific in a unit test that responded to link status, but I was unable to configure the nodes in such a way to produce the right configuration. However, this is because such a configuration is forbidden under our current rules, so it didn't need testing!

I think maybe it's OK, and in the general case you can always create equivalent "types" of behaviour within the grid using closed nodes as the main element instead of inactive links per se. An example might be a flow obstruction in the middle of the grid, which could be achieved either by severing individual links, or by adding a line of closed nodes. The only issue would be that the closed nodes equivalent would be wider - which is why this came up in a unit test context, where to make exotic geometries with just nodes you have to make your testing grid bigger...
